### ISSUE TITLE
AndroidChannelBuilderTest is incompatible with Robolectric 4.4

### ISSUE THREAD

**Issue by ejona86** [grpc/grpc-java#7731]
When I tried to upgrade to robolectric 4.4, compilation failed:

```
grpc-java/android/src/test/java/io/grpc/android/AndroidChannelBuilderTest.java:116: error: cannot access RoleManager
    shadowOf(connectivityManager).setActiveNetworkInfo(WIFI_CONNECTED);
    ^
  class file for android.app.role.RoleManager not found
1 error
```

It's not obvious to me what is going on, but I'm not familiar with `shadowOf()`

**Comment by markb74** [comment]
According to https://github.com/robolectric/robolectric/issues/5636, this should have been resolved when the Android API was bumped to 29.

(And it appears to work for me).

**Comment by ejona86** [comment]
Fixed by #7796.
